### PR TITLE
work to converge contract readers [4/n]

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

Creates a new folder for ContractReader and updates the v1 and v2 hooks to point to it with their respective contracts

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1869** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1869" target="_blank">Graphite</a> 
    * **PR #1870** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1870" target="_blank">Graphite</a> 
      * **PR #1871** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1871" target="_blank">Graphite</a> 
        * **PR #1878** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1878" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1878?utm_source=stack-comment).

File: src/hooks/v2/contractReader/V2ContractReader.ts [hunk 1]
```diff
@@ -1,92 +1,13 @@
...
-}: {
-  contract?: ContractConfig
-  functionName?: string
-  args?: unknown[] | null
-  updateOn?: ContractUpdateOn
-  formatter?: (val?: any) => V | undefined // eslint-disable-line @typescript-eslint/no-explicit-any
-  callback?: (val?: V) => void
-  valueDidChange?: (oldVal?: V, newVal?: V) => boolean
-}): ContractReadResult<V> {
+export default function useV2ContractReader<V>(
```
Comment (peripheralist) [inline src/hooks/v2/contractReader/V2ContractReader.ts:8]: wdyt about further abstracting the contractReader to be entirely version agnostic, and accept a ContractConfig that could be either V1 or V2?

the benefit would just be simplifying our hooks. not sure if there are any significant downsides when it comes to implementing

contracts are still read through the versioned V1/V2UserContext, so seems unlikely that a component would ever call the wrong version of a contract

Comment (wraeth-eth) [reply src/hooks/v2/contractReader/V2ContractReader.ts:8]: We technically can do this (it is just using the contract reader defined in this PR directly), but I don't know if the refactoring of all the code is worth it. It isn't a big job, more just tedious. 

Keeping it separate also opens our options for flexibility down the track (if we ever needed to add something V2/Vn specific) in the future

Comment (peripheralist) [reply src/hooks/v2/contractReader/V2ContractReader.ts:8]: ya makes sense 👍 if we do wanna refactor it that far we can do it in a different PR

**Pull request closed without merging**